mkt/tags/models.py

```diff
@@ -37,25 +37,22 @@ def can_reverse(self):
     def get_url_path(self):
         return reverse('tags.detail', args=[self.tag_text])
 
-    def save_tag(self, addon):
-        tag, created = Tag.objects.get_or_create(tag_text=self.tag_text)
-        AddonTag.objects.get_or_create(addon=addon, tag=tag)
-        mkt.log(mkt.LOG.ADD_TAG, tag, addon)
-        return tag
+    def _get_m2m_name(self, obj):
+        related_models = Tag._meta.get_all_related_m2m_objects_with_model()
+        field_map = dict((rm[0].model, rm[0].field.name)
```

Comment:
Could use a dict comprehension instead of calling `dict()` here since we are on python 2.7

Also, do you think you could also use this method (making it a `classmethod`) in `attach_tags()` below, removing the need to pass the `m2m_name` and determining it automatically instead ?